lib/ansible/galaxy/collection.py

```diff
@@ -0,0 +1,265 @@
+
+
+def _build_files_manifest(collection_path):
+    ignore_files = frozenset(['*.pyc', '*.retry'])
+    ignore_dirs = frozenset(['CVS', '.bzr', '.hg', '.git', '.svn', '__pycache__', '.tox'])
+
+    entry_template = {
+        'name': None,
+        'ftype': None,
+        'chksum_type': None,
+        'chksum_sha256': None,
+        '_format': MANIFEST_FORMAT
+    }
+    manifest = []
+
+    def _walk(path, top_level_dir, parent_dirs):
+        for item in os.listdir(path):
+            abs_path = os.path.join(path, item)
+            rel_base_dir = '' if path == top_level_dir else path[len(top_level_dir) + 1:]
+            rel_path = os.path.join(rel_base_dir, item)
+
+            if os.path.isdir(abs_path):
+                if item in ignore_dirs:
+                    display.vvv("Skipping %s for collection build" % abs_path)
+                    continue
+
+                if os.path.islink(abs_path):
+                    link_target = os.path.realpath(abs_path)
+
+                    if not link_target.startswith(top_level_dir):
+                        display.warning("Skipping %s as it is a symbolic link to a directory outside the collection"
+                                        % abs_path)
+                        continue
+
+                manifest_entry = entry_template.copy()
+                manifest_entry['name'] = rel_path
+                manifest_entry['ftype'] = 'dir'
+
+                manifest.append(manifest_entry)
+
+                _walk(abs_path, top_level_dir, parent_dirs)
+            else:
+                if item == 'galaxy.yml':
+                    continue
+                elif any(fnmatch.fnmatch(item, pattern) for pattern in ignore_files):
+                    display.vvv("Skipping %s for collection build" % abs_path)
+                    continue
+
+                manifest_entry = entry_template.copy()
+                manifest_entry['name'] = rel_path
+                manifest_entry['ftype'] = 'file'
+                manifest_entry['chksum_type'] = 'sha256'
+                manifest_entry['chksum_sha256'] = secure_hash(abs_path, hash_func=sha256)
+
+                manifest.append(manifest_entry)
+
+    dir_stats = os.stat(collection_path)
+    parents = frozenset(((dir_stats.st_dev, dir_stats.st_ino),))
+    _walk(collection_path, collection_path, parents)
+
+    return manifest
+
+
+def _build_manifest(namespace, name, version, authors, tags, description, license_ids, dependencies, **kwargs):
+
+    def _parse_author_info(author):
+        email = None
+        url = None
+
+        match = AUTHOR_REGEX.match(author)
+        if match:
+            author = match.group('name')
+            email = match.group('email')
+            url = match.group('url')
+
+        return {
+            'name': author,
+            'email': email,
+            'url': url
+        }
+    authors = [_parse_author_info(info) for info in authors]
+
+    deps = {}
+    for name, version in dependencies.items():
```

Comment:
```suggestion
    deps = {name: {'version': version} for name, version in dependencies.items()}
```

Reply:
Won't work due to dict comprehension not being supported in Python 2.6 and it will fail the sanity checks. Anyway this format isn't needed anymore as I was originally working on older docs.